apis/python/tests/test_lossy_ingest_outgest_roundtrips.py

```diff
@@ -0,0 +1,235 @@
+        make_df("xx,yy,zz", col0="AA,BB,CC", obs_id="aa,bb,cc"),
+        # ⇒ `obs_id` column becomes `df.index`, loses name
+        # ⇒ Original `df.index` dropped
+        make_df("aa,bb,cc", col0="AA,BB,CC"),
+        [ "col0", "obs_id", ],
+    ),
+    RoundTrip(
+        '3. DataFrame has a column named "index", and `df.index` is unnamed',
+        make_df("xx,yy,zz", col0="aa,bb,cc", index="AA,BB,CC"),
+        # ⇒ "index" column is promoted to `df.index` (unnamed)
+        # ⇒ Original (unnamed) index becomes a column named `level_0`
+        make_df("AA,BB,CC", level_0="xx,yy,zz", col0="aa,bb,cc"),
+        [ "level_0", "col0", "obs_id", ],
+    ),
+    RoundTrip(
+        '4. DataFrame has a column named "index", and `df.index` is named `id_column_name` (default `obs_id`)',
+        make_df("obs_id=xx,yy,zz", col0="aa,bb,cc", index="AA,BB,CC"),
+        # ⇒ "index" column is dropped
+        make_df("obs_id=xx,yy,zz", col0="aa,bb,cc"),
+        [ "obs_id", "col0", ],
+        "obs_id",
+    ),
+    RoundTrip(
+        '5. DataFrame has a column named "index" and df.index has another name',
+        make_df("idx=xx,yy,zz", col0="aa,bb,cc", index="AA,BB,CC"),
+        # ⇒ "index" column renamed to `obs_id`
+        make_df("idx=xx,yy,zz", col0="aa,bb,cc", obs_id="AA,BB,CC"),
+        [ "idx", "col0", "obs_id", ],
+        "idx",
+    ),
+    RoundTrip(
+        '6. DataFrame has columns named "index" and `id_column_name` (default: obs_id), and `df.index` is unnamed:',
+        make_df("xx,yy,zz", obs_id="aa,bb,cc", index="AA,BB,CC"),
+        # ⇒ unnamed index → column named `level_0`
+        # ⇒ `obs_id` column → unnamed index
+        # ⇒ "index" column dropped
+        make_df("aa,bb,cc", level_0="xx,yy,zz"),
+        [ "level_0", "obs_id", ],
+    ),
+    RoundTrip(
+        '7. DataFrame has columns named "index" and `id_column_name` (default: obs_id), and `df.index` has a name:',
+        make_df("idx=xx,yy,zz", obs_id="aa,bb,cc", index="AA,BB,CC"),
+        # ⇒ "index" column is dropped
+        make_df("idx=xx,yy,zz", obs_id="aa,bb,cc"),
+        [ "idx", "obs_id", ],
+        "idx",
+    ),
+]
+# fmt: on
+
+
+def verify_metadata(
+    sdf: DataFrame, persisted_cols: List[str], persisted_metadata: Optional[str]
+):
+    # Verify column names and types
+    schema = sdf.schema
+    assert schema.names == [SOMA_JOINID, *persisted_cols]
+    [soma_joinid_type, *string_col_types] = schema.types
+    assert soma_joinid_type == pa.int64() and schema.field(0).nullable is False
+    for string_col_type in string_col_types:
+        assert string_col_type == pa.large_string()
+
+    # Verify "original index metadata"
+    actual_index_metadata = json.loads(
+        sdf.metadata[_DATAFRAME_ORIGINAL_INDEX_NAME_JSON]
+    )
+    assert actual_index_metadata == persisted_metadata
+
+
+@parametrize_roundtrips(ROUND_TRIPS)
+def test_lossy_roundtrips_adata(
+    tmp_path: Path,
+    original_df: pd.DataFrame,
+    persisted_cols: List[str],
```

Comment:
```suggestion
    persisted_column_names: List[str],
```
and likewise thoughout this file.

A column is a named array. A column name is a string. Since this is `List[str]`, it's a list of column names, and should be called `persisted_column_names`.